pkg/redo/writer/file/file.go

```diff
@@ -229,6 +247,42 @@ func (w *Writer) IsRunning() bool {
 	return w.running.Load()
 }
 
+func (w *Writer) GetInputCh() chan writer.RedoEvent {
+	return w.inputCh
+}
+
+func (w *Writer) SyncWrite(event writer.RedoEvent) error {
+	rl := event.ToRedoLog()
+	data, err := codec.MarshalRedoLog(rl, nil)
+	if err != nil {
+		return errors.WrapError(errors.ErrMarshalFailed, err)
+	}
+	w.AdvanceTs(rl.GetCommitTs())
+	_, err = w.Write(data)
+	if err != nil {
+		return err
+	}
+	err = w.Flush()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	event.PostFlush()
+	return nil
+}
+
+func (w *Writer) encode(ctx context.Context) error {
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case e := <-w.inputCh:
+		err := w.SyncWrite(e)
```

Comment:
So we receive each event, and will write it without a batch action? Will it cost some performance problem.